### PR TITLE
Integrated test.sh in Makefile

### PR DESCRIPTION
**Pull request by matfra**
Now make test is a little bit smarter. It doesn't systematically delete the .paasta virtualenv
Non primary runs of make test now takes 8 seconds on my laptop.

### CODE REVIEW

**Comment by matfra** [comment]
Oops, python 3.7 is not gonna work ;-)

**Comment by matfra** [comment]
Integrated itest.sh and docs.sh to Makefile too

File: Makefile [hunk 1]
```diff
@@ -22,14 +22,22 @@ endif
 
 .PHONY: all docs test itest
 
-docs:
-	./docs.sh
+docs: .paasta/bin/activate
```
Comment (solarkennedy) [inline Makefile]: Can you ship these makefile changes in a different PR so they can be bike-sheded independently?

Comment (solarkennedy) [inline Makefile]: Wait sorry this *is* a different PR :)

File: Makefile [hunk 2]
```diff
@@ -22,14 +22,22 @@ endif
...
-test:
-	./test.sh
+test: .paasta/bin/activate
+	.paasta/bin/tox
 
-itest: test
-	./itest.sh
+.paasta/bin/activate: requirements.txt requirements-dev.txt
+	test -d .paasta/bin/activate || virtualenv -p python2.7 .paasta
+	.paasta/bin/pip install -U pip
```
Comment (drolando) [inline Makefile]: you should pin the pip virtualenv and tox versions

**Comment by matfra** [comment]
Pinned version as recommended by @drolando . But is it really necessary?

**Comment by mattmb** [comment]
I've been meaning to pin the versions for a while actually, could you do me a favour and have a grep for anywhere else we're installing pip/virtualenv etc. and pin them too. I'm pretty sure there's some stuff in the various Dockerfiles.

Thanks for cleaning this up, I made a bit of a mess when I spent a few days hitting it with a hammer until the builds went green!

**Comment by matfra** [comment]
@mattmb I've pinned the version as you requested. I left itest.sh intact has make itest won't finish properly on my mac without it (this is because by default itest.sh is doesn't exit on failure and Makefile does)
Also I rebased on the latest master. I haven't tried to make release

**Pull request merged**